prow/spyglass/artifactfetchers.go

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package spyglass
+
+import (
+	"errors"
+
+	"k8s.io/test-infra/prow/spyglass/viewers"
+)
+
+var (
+	// ErrCannotParseSource is thrown when an invalid source string is provided to an ArtifactFetcher
+	ErrCannotParseSource = errors.New("could not create job source from provided source")
+)
+
+// jobSource gets information about a location storing the results of a single Prow job
+type jobSource interface {
+	// CanonicalLink returns a link to the storage location this job's artifacts.
+	CanonicalLink() string
```

Comment:
Does an internal interface need exported methods?

It is also not at all obvious to me why we need to define the interface here as opposed to letting the consumer define the interface. Interface I does not need to be defined when authoring package X. Package Y can just later define the interface and if X implements those methods all is good.